src/compile/data/assemble.ts

```diff
@@ -156,6 +156,19 @@ function makeWalkTree(data: VgData[]) {
    * Recursively walk down the tree.
    */
   function walkTree(node: DataFlowNode, dataSource: VgData) {
+    if (node instanceof ParseNode) {
+      if (node.parent instanceof SourceNode && !node.parent.parent)  {
```

Comment:
drop `&& !node.parent.parent` as a source node has to be a root. I will add code to enforce that in source nodes.